android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt

```diff
@@ -135,6 +135,36 @@ fun Patient.extractAddress(): String {
   }
 }
 
+fun Patient.extractAddressDistrict(): String {
+  if (!hasAddress()) return ""
+  return with(addressFirstRep) { this.district ?: "" }
+}
+
+fun Patient.extractAddressState(): String {
+  if (!hasAddress()) return ""
+  return with(addressFirstRep) { this.state ?: "" }
+}
+
+fun Patient.extractAddressText(): String {
+  if (!hasAddress()) return ""
+  return with(addressFirstRep) { this.text ?: "" }
+}
+
+fun Patient.extractTelecom(): List<String>? {
+  if (!hasTelecom()) return null
+  return telecom.map { it.value }
+}
```

Comment:
It's possible to have these implementation in one function and use utility class `org.smartregister.fhircore.engine.util.fhirpath.FhirPathDataExtractor` to extract the relevant properties from the patient resource.